### PR TITLE
Fix bug #1177 Sidepanel stat icons flickers during turn updates.

### PR DESCRIPTION
**Pull request by LGM-Doyle**
This fixes issue #1177.

The underlying problem was that the `StatisticIcons` for the Military, Population and Resource panels were initialized with zeros instead of the correct value, and then immediately rendered.

This PR:
- uses the correct initial values for the panel icons.
- adds `StatisticIcon::MinUsableSize()` so that the icons can be clipped as tightly as possible
- adds `PreRender()` to the panels so that the icons are only layed out once.
- moves the rendering, positioning and layering of the `AccordionPanel` expand button into `AccordionPanel`

### CODE REVIEW

**Review by Vezzra — APPROVED** [review]
Yep, fixed the problem, the flickering is gone (tested on OSX). Can't say much about the code, looks fine to me.

**Comment by Vezzra** [comment]
Corresponding issue #1177 is assigned to "optional for 0.4.7" milestone, so should a PR addressing it.

File: UI/MilitaryPanel.cpp [hunk 1]
```diff
@@ -41,16 +41,27 @@ MilitaryPanel::MilitaryPanel(GG::X w, int planet_id) :
...
 
+    const auto obj = GetUniverseObject(m_planet_id);
+    if (!obj) {
+        ErrorLogger() << "Invalid object id " << m_planet_id;
+        return;
+    }
+
     // small meter indicators - for use when panel is collapsed
-    m_meter_stats.push_back({METER_SHIELD, new StatisticIcon(ClientUI::MeterIcon(METER_SHIELD), 0, 3, false,
+    m_meter_stats.push_back({METER_SHIELD, new StatisticIcon(ClientUI::MeterIcon(METER_SHIELD),
```
Comment (geoffthemedio) [inline UI/MilitaryPanel.cpp:51]: are these changes related to the point of this pull request?

Comment (LGM-Doyle) [reply UI/MilitaryPanel.cpp:51]: The values that the icon was flickering between were the initial incorrect default value and the correct value.  These lines initialize the icons with the correct value.

**Comment by geoffthemedio** [comment]
haven't tested myself, but no objections
